android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIButton.java

```diff
@@ -73,24 +80,51 @@ public TiUIButton(final TiViewProxy proxy)
 				break;
 		}
 
-		MaterialButton btn = new MaterialButton(proxy.getActivity(), null, styleId) {
-			@Override
-			public boolean onFilterTouchEventForSecurity(MotionEvent event)
-			{
-				boolean isTouchAllowed = super.onFilterTouchEventForSecurity(event);
-				if (!isTouchAllowed) {
-					fireSyncEvent(TiC.EVENT_TOUCH_FILTERED, dictFromEvent(event));
-				}
-				return isTouchAllowed;
-			}
+		// Determine if a background drawable will be applied to the button.
+		boolean hasCustomBackground
+			=  hasImage(proxy.getProperties())
+			|| hasColorState(proxy.getProperties())
+			|| hasBorder(proxy.getProperties())
+			|| hasGradient(proxy.getProperties());
 
+		// Create and set up the button.
+		// Note: MaterialButton does not support replacing its background drawable. Will log a nasty warning.
+		AppCompatButton btn;
+		if (hasCustomBackground) {
+			btn = new AppCompatButton(proxy.getActivity()) {
+				@Override
+				public boolean onFilterTouchEventForSecurity(MotionEvent event)
```

Comment:
Potential to move this out and remove duplicate code?

Reply:
I thought about it, but the duplicated code was so small that I didn't think it was worth it. Especially since it would bloat it to the below. 🙄 
```java
public boolean onFilterTouchEventForSecurity(MotionEvent event)
{
	boolean isTouchAllowed = super.onFilterTouchEventForSecurity(event);
	return handleFilterTouchEventForSecurity(isTouchAllowed, event);
}

private boolean handleFilterTouchEventForSecurity(boolean isAllowed, MotionEvent event)
{
	if (!isAllowed) {
		fireSyncEvent(TiC.EVENT_TOUCH_FILTERED, dictFromEvent(event));
	}
	return isAllowed;
}
```